app/src/main/java/com/android/voyageur/ui/authentication/SignIn.kt

```diff
@@ -40,23 +39,61 @@ import com.google.firebase.auth.auth
 import kotlinx.coroutines.launch
 import kotlinx.coroutines.tasks.await
 
+@Composable
+private fun rememberFirebaseAuthLauncher(
+    onAuthComplete: (AuthResult) -> Unit,
+    onAuthError: (ApiException) -> Unit
+): ManagedActivityResultLauncher<Intent, ActivityResult> {
+  val scope = rememberCoroutineScope()
+
+  return rememberLauncherForActivityResult(ActivityResultContracts.StartActivityForResult()) {
+      result ->
+    val task = GoogleSignIn.getSignedInAccountFromIntent(result.data)
+    try {
+      val account = task.getResult(ApiException::class.java)!!
+      val credential = GoogleAuthProvider.getCredential(account.idToken!!, null)
+      scope.launch {
+        val authResult = Firebase.auth.signInWithCredential(credential).await()
+        onAuthComplete(authResult)
+      }
+    } catch (e: ApiException) {
+      onAuthError(e)
+    }
+  }
+}
+
+@Composable
+fun AuthenticationWrapper(navigationActions: NavigationActions) {
+  val user by remember { mutableStateOf(Firebase.auth.currentUser) }
+
+  LaunchedEffect(user) {
+    if (user != null) {
+      navigationActions.navigateTo(TopLevelDestinations.OVERVIEW)
+    }
+  }
+
+  if (user == null) {
+    SignInScreen(navigationActions)
+  }
+}
+
 @Composable
 fun SignInScreen(navigationActions: NavigationActions) {
```

Comment:
Could you make the screen scrollable, e.g. if I have my phone in landscape I can't connect to the app.